contrib/platform/src/com/sun/jna/platform/unix/X11.java

```diff
@@ -797,6 +797,8 @@ void XSetWMProperties(Display display, Window window, String window_name,
                           Pointer class_hints);
 
     int XSetWMProtocols(Display display, Window window, Atom[] atom, int count);
+    int XGetWMProtocols(Display display, Window w, Atom[] protocols_return, IntByReference count_return);
```

Comment:
This won't work - the function is declared as:

```c
Status XGetWMProtocols(Display *display, Window w, Atom
    **protocols_return, int *count_return); 
```

protocols_return is a pointer to an array of atoms (following the documentation). The array of atoms is allocated on the callee side and not on the java side. As an Atom is basicly a primitive long, I only see the option to bind this as a PointerByReference. From that you have to extract the data itself and in the end free the list:

```java
        X11.Atom[] atomList = new X11.Atom[count.getValue()];
        for(int i = count.getValue() - 1; i >= 0; i--) {
            atomList[i] = new X11.Atom(protocols.getValue().getLong(X11.Atom.SIZE * i));
        }
        X11.INSTANCE.XFree(protocols.getValue());
```